ql/src/test/results/clientpositive/llap/dynamic_semijoin_reduction_2.q.out

```diff
@@ -79,27 +79,25 @@ STAGE PLANS:
     Tez
 #### A masked pattern was here ####
       Edges:
```

Comment:
Interesting: New plan seems more compact (less stages), which is good. I am wondering if this has to do with Shared Work Optimizer finding further optimization opportunities? Or maybe something else? Have you checked?

Reply:
The plan in `dynamic_semijoin_reduction_2.q` has three single column semijoin reducers that get merged to one multi column one. As a result three reducers get merged to one thus making the plan more compact.  

Apart from that, you are right that the multi column transformation can lead to further optimization opportunities. An example can be seen in query24.q.out (Check commit https://github.com/apache/hive/pull/1325/commits/c9f9112d0802906dce7442f3d4c01535a584af11). There the `SharedWorkOptimizer` kicks in and merges two semijoin reducer branches on the same scan operator.